deno/lib/types.ts

```diff
@@ -1004,38 +1004,91 @@ export class ZodBoolean extends ZodType<boolean, ZodBooleanDef> {
 //////////                     ////////
 ///////////////////////////////////////
 ///////////////////////////////////////
+type ZodDateCheck =
+  | { kind: "min"; value: number; message?: string }
+  | { kind: "max"; value: number; message?: string };
 export interface ZodDateDef extends ZodTypeDef {
+  checks: ZodDateCheck[];
   typeName: ZodFirstPartyTypeKind.ZodDate;
 }
 
 export class ZodDate extends ZodType<Date, ZodDateDef> {
   _parse(input: ParseInput): ParseReturnType<this["_output"]> {
     const parsedType = this._getType(input);
+    const ctx: undefined | ParseContext = this._getOrReturnCtx(input);
+
     if (parsedType !== ZodParsedType.date) {
-      const ctx = this._getOrReturnCtx(input);
       addIssueToContext(ctx, {
         code: ZodIssueCode.invalid_type,
         expected: ZodParsedType.date,
         received: ctx.parsedType,
       });
       return INVALID;
     }
+
     if (isNaN(input.data.getTime())) {
-      const ctx = this._getOrReturnCtx(input);
       addIssueToContext(ctx, {
         code: ZodIssueCode.invalid_date,
       });
       return INVALID;
     }
 
+    const status = new ParseStatus();
+
+    for (const check of this._def.checks) {
+      if (check.kind === "min") {
+        if (input.data.getTime() < check.value) {
```

Comment:
In your error message, you hard-coded `or equal to`, but here you use `<`. For dates, it would (in my opinion) make more sense if `min` and `max` were inclusive. So I would switch this to `<=` and add `inclusive: true` to the issue generated if it's not valid.